invenio_accounts/sessions.py

```diff
@@ -56,10 +54,10 @@ def add_session(session=None):
     user_id, sid_s = session['user_id'], session.sid_s
     session_activity = SessionActivity(user_id=user_id, sid_s=sid_s)
     try:
-        _datastore.db.session.add(session_activity)
-        _datastore.commit()
+        _db.session.add(session_activity)
```

Comment:
So using `with begin_nested()` we're safe against exceptions thrown by `db.session.commit`. 
However exceptions thrown by statements inside the `with` block still need to be handled. Integrity Errors, for example, are thrown immediately if you try to add a duplicate entry to the session, not when you commit.